### PR TITLE
Update CHANGELOG for release v0.10

### PR DESCRIPTION
**Pull request by co9olguy**

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -34,10 +36,15 @@
...
-* Contains the new template `UCCSD` implementing the Unitary Coupled-Cluster (UCCSD) ansatz
-  to perform VQE-based quantum chemistry simulations using PennyLane-QChem.
-  [(#654)](https://github.com/XanaduAI/pennylane/pull/654)
-
+* The ``default.tensor`` plugin has been significantly upgraded. It now allows two different
+  tensor network representations to be used: `"exact"` and `"mps"`. The former uses a 
+  exact factorized representation of quantum states, while the latter uses a matrix product state 
+  representation.
+  ([#572](https://github.com/XanaduAI/pennylane/pull/572))
+  ([#599](https://github.com/XanaduAI/pennylane/pull/599))
```
Comment (josh146) [inline .github/CHANGELOG.md:44]: 💯

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -2,9 +2,11 @@
 
 <h3>New features since last release</h3>
 
-* Adds a new device, `default.qubit.tf`, a pure-state qubit simulator written using TensorFlow.
+<h4>New and improved simulators</h4>
```
Comment (josh146) [inline .github/CHANGELOG.md:5]: I like the subsection headings :+1:

File: .github/CHANGELOG.md [hunk 3]
```diff
@@ -53,7 +60,7 @@
   weight_shapes = {"weights_0": 3, "weight_1": 1}
   qlayer = qml.qnn.TorchLayer(qnode, weight_shapes)
   ```
-
+
```
Comment (josh146) [inline .github/CHANGELOG.md:63]: Lol I tried to modify this using GitHub's 'suggested changes' option on the PR a while back, and gave up --- GitHub struggled to deal with a suggested change, when the suggested change involved manipulating three backticks 😆

File: .github/CHANGELOG.md (hunk 1 above)
Comment (josh146) [reply .github/CHANGELOG.md:44]: Maybe should link to the `default.tensor` docstring, since no example is given here?

File: .github/CHANGELOG.md [hunk 4]
```diff
@@ -62,65 +69,29 @@
 * Added a new "reversible" differentiation method which can be used in simulators, but not hardware.
   The reversible approach is similar to backpropagation, but trades off extra computation for
   enhanced memory efficiency. Where backpropagation caches the state tensors at each step during
-  a forward pass, the reversible method only caches the final pre-measurement state. Compared to 
+  a simulated evolution, the reversible method only caches the final pre-measurement state. Compared to 
   the parameter-shift method, the reversible method can be faster or slower, depending on the 
   density and location of parametrized gates in a circuit (circuits with higher density of 
   parametrized gates near the end of the circuit will see a benefit).
   [(#670)](https://github.com/XanaduAI/pennylane/pull/670)
```
Comment (josh146) [inline .github/CHANGELOG.md:76]: Would recommend either adding a short example of usage here, something like
```python
@qml.qnode(dev, diff_method="reversible")
def circuit(params):
    ...
```
just to show users how it is accessed from the UI (it's not very clear from the changelog entry).

File: .github/CHANGELOG.md [hunk 5]
```diff
@@ -62,65 +69,29 @@
...
 
-* Contains the new template `DoubleExcitationUnitary` implementing the quantum circuit to
-  exponentiate the Coupled-Cluster double excitation operator. This template is required to
-  build the Unitary Coupled-Cluster Singles and Doubles (UCCSD) ansatz for VQE simulations.
+* Added the new templates `UCCSD`, `SingleExcitationUnitary`, and`DoubleExcitationUnitary`,
+  which together implement the Unitary Coupled-Cluster Singles and Doubles (UCCSD) ansatz
+  to perform VQE-based quantum chemistry simulations using PennyLane-QChem.
+  [(#654)](https://github.com/XanaduAI/pennylane/pull/654)
   [(#638)](https://github.com/XanaduAI/pennylane/pull/638)
   [(#659)](https://github.com/XanaduAI/pennylane/pull/659)
```
Comment (josh146) [inline .github/CHANGELOG.md:85]:
```suggestion
  [(#622)](https://github.com/XanaduAI/pennylane/pull/622)
  [(#638)](https://github.com/XanaduAI/pennylane/pull/638)
  [(#654)](https://github.com/XanaduAI/pennylane/pull/654)
  [(#659)](https://github.com/XanaduAI/pennylane/pull/659)
```

File: .github/CHANGELOG.md [hunk 6]
```diff
@@ -34,10 +36,15 @@
   [documentation](https://pennylane.ai/en/stable/code/api/pennylane.beta.plugins.DefaultQubitTF.html)
   for more details.
 
-* Contains the new template `UCCSD` implementing the Unitary Coupled-Cluster (UCCSD) ansatz
-  to perform VQE-based quantum chemistry simulations using PennyLane-QChem.
-  [(#654)](https://github.com/XanaduAI/pennylane/pull/654)
-
+* The ``default.tensor`` plugin has been significantly upgraded. It now allows two different
```
Comment (co9olguy) [inline .github/CHANGELOG.md:39]:
```suggestion
* The [``default.tensor`` plugin](https://github.com/XanaduAI/pennylane/blob/master/pennylane/beta/plugins/default_tensor.py) has been significantly upgraded. It now allows two different
```

File: .github/CHANGELOG.md [hunk 7]
```diff
@@ -62,65 +69,42 @@
...
+  
+  ```pycon
+  >>> dev = qml.device("default.qubit", wires=2)
+  ... @qml.qnode(dev, diff_method="reversible")
+  ... def circuit(x):
+  ...     qml.RX(x, wires=0)
+  ...     qml.RX(x, wires=0)
+  ...     qml.CNOT(wires=[0,1])
+  ...     return qml.expval(qml.PauliZ(0))
+  >>> qml.grad(circuit)(0.5, 1.2)
```
Comment (co9olguy) [inline .github/CHANGELOG.md:86]:
```suggestion
  >>> qml.grad(circuit)(0.5)
```

**Comment by co9olguy** [comment]
@josh146 ready to go

File: .github/CHANGELOG.md [hunk 8]
```diff
@@ -36,7 +36,7 @@
   [documentation](https://pennylane.ai/en/stable/code/api/pennylane.beta.plugins.DefaultQubitTF.html)
   for more details.
 
-* The ``default.tensor`` plugin has been significantly upgraded. It now allows two different
+* The [``default.tensor`` plugin](https://github.com/XanaduAI/pennylane/blob/master/pennylane/beta/plugins/default_tensor.py) has been significantly upgraded. It now allows two different
```
Comment (josh146) [inline .github/CHANGELOG.md:39]: Even thought Markdown does support nested formatting, ReST doesn't :(
```suggestion
* The [default.tensor plugin](https://github.com/XanaduAI/pennylane/blob/master/pennylane/beta/plugins/default_tensor.py) has been significantly upgraded. It now allows two different
```

File: .github/CHANGELOG.md [hunk 9]
```diff
@@ -51,37 +51,52 @@
...
-  ```python
-  @qml.qnode(dev)
-  def qnode(inputs, weights_0, weight_1):
-     # define the circuit
-     # ...
+  ```pycon
+  >>> @qml.qnode(dev)
+  ... def qnode(inputs, weights_0, weight_1):
+  ...    # define the circuit
+  ...    # ...
```
Comment (josh146) [inline .github/CHANGELOG.md:59]:
```suggestion
  ...    # ...
```

File: .github/CHANGELOG.md [hunk 10]
```diff
@@ -51,37 +51,52 @@
...
+  ```pycon
+  >>> dev = qml.device("default.qubit", wires=2)
+  ... @qml.qnode(dev, diff_method="reversible")
+  ... def circuit(x):
+  ...     qml.RX(x, wires=0)
+  ...     qml.RX(x, wires=0)
+  ...     qml.CNOT(wires=[0,1])
+  ...     return qml.expval(qml.PauliZ(0))
+  >>> qml.grad(circuit)(0.5)
+  (array(-0.47942554),)
```
Comment (josh146) [inline .github/CHANGELOG.md:89]: 💯

File: .github/CHANGELOG.md (hunk 9 above)
Comment (co9olguy) [reply .github/CHANGELOG.md:59]: lol this "suggestion" is not registering as something that is committable in github

**Pull request merged**